src/services/output_parsers.py

```
@@ -0,0 +1,35 @@
+from langchain_core.output_parsers import JsonOutputParser
+
+
+class RobustJsonOutputParser(JsonOutputParser):
+    """JSON Output Parser that can handle text with thinking tags and extra content"""
+
+    def parse(self, text: str) -> dict:
+        """Parse JSON from text, handling <think> tags and other artifacts"""
+        try:
+            # First try standard JSON parsing
+            return super().parse(text)
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)